class_15_sockets/homework/server.py

```diff
@@ -0,0 +1,78 @@
+import pathlib
+import selectors
+import socket
+from collections import defaultdict
+from multiprocessing import Lock
+from multiprocessing.pool import ThreadPool
+
+HOST = "localhost"
+PORT = 9999
+
+selector = selectors.DefaultSelector()
+pool = ThreadPool(4)
+bufs = defaultdict(bytes)
+clients = defaultdict(str)
+lock = Lock()
+
+
+def append_data(data):
+    data = data.decode()
+    pathlib.Path("outputs").mkdir(parents=True, exist_ok=True)
+    with lock:
```

Comment:
👍 for using `Lock`, because in the case of several clients we will have problems with writing in one file

Reply:
thank you